File: app/src/main/java/com/example/engineeraitest/app/EngineerAiTestApp.java
package com.example.engineeraitest.app;

import android.app.Application;

import com.example.engineeraitest.app.di.DependencyGraph;

public class EngineerAiTestApp extends Application {

    private static DependencyGraph mDependencyGraph;

    @Override
    public void onCreate() {
        super.onCreate();
        mDependencyGraph = new DependencyGraph(this);
    }

    public static DependencyGraph getDependencyGraph() {
        return mDependencyGraph;
    }
}

File: app/src/main/java/com/example/engineeraitest/app/EngineerAiTestApp.kt
package com.example.engineeraitest.app

import android.app.Application
import com.example.engineeraitest.app.di.appModules
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin

class EngineerAiTestApp : Application() {

    override fun onCreate() {
        super.onCreate()
        startKoin {
            androidContext(this@EngineerAiTestApp)
            modules(appModules)
        }
    }
}